### PR TITLE
enhancement(remap): Add to_syslog_severity function for Syslog log level keywords

### PR DESCRIPTION
**Pull request by lucperkins**
Addresses #5501 with a `to_syslog_severity` Remap function that converts a Syslog log level keyword into an integer severity level.

**Note**: the heavy lifting has been done here. I'm marking this as a draft until (a) #5522 lands and (b) we figure out a naming system for modules/namespaces.

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
Maybe we should call this `to_syslog_severity`? This would be more specific and leave the door open for other severity mappings.

**Comment by jszwedko** [comment]
Thttps://github.com/timberio/vector/issues/5507 seems related which proposes calling this `syslog::to_severity`.

**Review by jszwedko — COMMENTED** [review]
In absence of #5507, which feels like it might be a little ways off, I'd propose calling this `to_syslog_severity`. We can always deprecate it and change to `syslog::to_severity` later.

Otherwise, this looks good to me!

File: docs/reference/remap/to_syslog_severity.cue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		`panic`, `error`, and `warn` are converted to `0`, `3`, and `4` respectively.
+		"""
+
+	examples: [
+		{
+			title: "Success"
+			input: {
+				string: "alert"
+			}
+			source: ".severity = to_severity(.log_level)"
```
Comment (jszwedko) [inline docs/reference/remap/to_syslog_severity.cue:27]:
```suggestion
			source: ".severity = to_syslog_severity(.log_level)"
```

File: lib/remap-functions/src/to_syslog_severity.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToSyslogSeverity;
+
+impl Function for ToSyslogSeverity {
+    fn identifier(&self) -> &'static str {
+        "to_severity"
```
Comment (jszwedko) [inline lib/remap-functions/src/to_syslog_severity.rs:8]:
```suggestion
        "to_syslog_severity"
```

File: lib/remap-functions/src/to_syslog_severity.rs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        let severity = match &level[..] {
+            "emerg" | "panic" => Ok(0),
+            "alert" => Ok(1),
+            "crit" => Ok(2),
+            "err" | "error" => Ok(3),
+            "warning" | "warn" => Ok(4),
+            "notice" => Ok(5),
+            "info" => Ok(6),
+            "debug" => Ok(7),
+            _ => Err(format!("level {} not valid", level)),
```
Comment (jszwedko) [inline lib/remap-functions/src/to_syslog_severity.rs:47]:
```suggestion
            _ => Err(format!("level `{}` is not a valid syslog level", level)),
```

File: lib/remap-functions/src/to_syslog_severity.rs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        };
+
+        match severity {
+            Ok(severity) => Ok(Value::from(severity)),
+            Err(e) => Err(Error::Call(e)),
```
Comment (jszwedko) [inline lib/remap-functions/src/to_syslog_severity.rs:52]: Nite: Just noting that in the other PR (#5503) the `Error::Call` was unneeded due to using ` Err(Error::from(...))` above. I don't have strong opinions on which one is better, but it'd be nice to be consistent.

**Review by jszwedko — APPROVED** [review]
A few last notes, otherwise this looks good.

File: lib/remap-functions/src/to_syslog_severity.rs (hunk 4 above)
Comment (lucperkins) [reply lib/remap-functions/src/to_syslog_severity.rs:52]: Good catch. Will fix.

**Pull request merged**